### PR TITLE
Improve test.sh

### PR DESCRIPTION
**Pull request by szepeviktor**
@johnbillion Shortening jq filter making the condition explicit: jq's output must be `[]`

### CODE REVIEW

File: tests/bin/test.sh [hunk 1]
```diff
@@ -5,25 +5,22 @@
...
+	fi
 }
 
 function validate_schema() {
 	local file="$1"
-	local base=${file//schemas\//}
-	local base=${base/.json/}
-	./node_modules/.bin/ajv validate -m tests/hyper-schema/hyper-schema.json -r "schemas/**/*.json" -s "$file" -d "tests/data/$base/*.json"
+	local base="$(basename "$file" .json)"
+	./node_modules/.bin/ajv validate -m tests/hyper-schema/hyper-schema.json -r "schemas/**/*.json" -s "$file" -d "tests/data/${base}/*.json"
```
Comment (johnbillion) [inline tests/bin/test.sh]: Pretty sure that using `basename` here means the path provided to the `-d` flag is incorrect for the REST API tests

Comment (szepeviktor) [reply tests/bin/test.sh]: I see! That "base" is not the basename.
```bash
$ basename schemas/rest-api/post.json .json
post
```

Comment (szepeviktor) [reply tests/bin/test.sh]: It should be "rest-api/post"

Comment (szepeviktor) [reply tests/bin/test.sh]: Why has `ajv validate` not failed??

Comment (szepeviktor) [reply tests/bin/test.sh]: Only `tests/data/rest-api/routes/*.json` files exist, and `schemas/rest-api/routes.json` does **not** exist.
What do you think?

Comment (johnbillion) [reply tests/bin/test.sh]: I'm working on some exceptions as we speak :) You might want to hold off making any further changes to this file while I work on that.

**Pull request closed without merging**